assests/javascript/index.js

```diff
@@ -5,23 +5,30 @@ const addTaskIcon = document.getElementById("add-task-icon");
 const addTaskBox = document.getElementById("add-task-box");
 const addTaskInputErrorMassage = document.getElementById("title-error");
 const networkErrorMassage = document.getElementById("network-error");
+const taskList = document.getElementById("task-list");
+const filterTasksView = document.getElementById("filter-tasks");
 
 addTaskInput.addEventListener(
   "keydown",
   (event) => {
     if (event.key === "Enter") {
       handleNewTask(addTaskInput.value);
+      addTaskInput.value = "";
     }
   }
 );
 
 addTaskIcon.addEventListener("click", () => {
   handleNewTask(addTaskInput.value);
+  addTaskInput.value = "";
 });
 
-async function handleNewTask(task) {
+showAllTasks();
+
+async function handleNewTask(newTask) {
   hideErrors();
-  addTask(task);
+  await addTask(newTask);
+  showAllTasks();
```

Comment:
It's not necessary to re-fetch all the tasks again in case that new task fails. right?

Reply:
yessss thanks